VogtObserver4/ResearchCenter.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VogtObserver4
+{
+    class ResearchCenter : RadiationMonitor
+    {
+        private string _location;
+        private double _average;
+        private int _count = 1;
+        public ResearchCenter(string location) : base(location)
+        {
+            _location = location;
+        }
+
+        public override string GenerateReport(double observation)
+        {
+            _count++;
+            _average += observation;
+            _average = _average / _count;
+
+            return Convert.ToString(_average);
+        }
+
+        public override void Update(IObservable subject, Object o)
```

Comment:
Looking at all of your child classes here, I think this function is 100% identical. That means you should be able to remove this from all child classes and put it on the parent class.

No need to make it abstract/virtual.

`GenerateReport` is the poylmorphic function which is different PER child class.

`Update()` is generic it looks like.